presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -142,54 +147,68 @@
 
     @Inject
     public GlueHiveMetastore(HdfsEnvironment hdfsEnvironment, GlueHiveMetastoreConfig glueConfig)
-    {
-        this(hdfsEnvironment, glueConfig, createAsyncGlueClient(glueConfig));
-    }
-
-    public GlueHiveMetastore(HdfsEnvironment hdfsEnvironment, GlueHiveMetastoreConfig glueConfig, AWSGlueAsync glueClient)
```

Comment:
I think you can set credential provider class name directly in `GlueHiveMetastoreConfig`. No need for passthrough via `Configuration`